app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -79,51 +81,60 @@ class GroupConversationDetailsViewModel @Inject constructor(
-                            }.collect { (conversationDetails, isSelfAnAdmin, selfTeam) ->
-                                with(conversationDetails) {
-                                    val isSelfInOwnerTeam =
-                                        selfTeam?.id != null && selfTeam.id == conversationDetails.conversation.teamId?.value
+        viewModelScope.launch(dispatcher.io()) {
+
+            val groupDetailsFlow = observeConversationDetails(conversationId)
+                .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>()
+                .map { it.conversationDetails }
+                .filterIsInstance<ConversationDetails.Group>()
+                .distinctUntilChanged()
+                .shareIn(this, SharingStarted.WhileSubscribed(), 1)
+
+            val isSelfAdminFlow = observeConversationMembers(conversationId)
+                .map { it.isSelfAnAdmin }
+                .distinctUntilChanged()
 
-                                    if (this is ConversationDetails.Group) {
-                                        updateState(
-                                            groupOptionsState.copy(
-                                                groupName = conversation.name.orEmpty(),
-                                                protocolInfo = conversation.protocol,
-                                                isUpdatingAllowed = isSelfAnAdmin,
-                                                areAccessOptionsAvailable = conversation.isTeamGroup(),
-                                                isGuestAllowed = (conversation.isGuestAllowed() || conversation.isNonTeamMemberAllowed()),
-                                                isServicesAllowed = conversation.isServicesAllowed(),
-                                                isUpdatingGuestAllowed = isSelfAnAdmin && isSelfInOwnerTeam,
-                                                isAbleToRemoveGroup = selfUser.teamId != null
-                                                        && conversation.creatorId.value == selfUser.id.value
-                                            )
-                                        )
-                                    }
-                                }
-                            }
+            launch {
+                groupDetailsFlow
+                    .collect { groupDetails ->
+                        with(groupDetails) {
+                            updateState(
+                                groupOptionsState.copy(
+                                    groupName = conversation.name.orEmpty(),
+                                    protocolInfo = conversation.protocol,
+                                    areAccessOptionsAvailable = conversation.isTeamGroup(),
+                                    isGuestAllowed = (conversation.isGuestAllowed() || conversation.isNonTeamMemberAllowed()),
+                                    isServicesAllowed = conversation.isServicesAllowed(),
```

Comment:
```suggestion
                                    areServicesAllowed = conversation.areServicesAllowed(),
```

Reply:
yes, but it requires changes in kalium 😓

Reply:
also IMO `is` is better here. As a developer, if I see some val start with `is` I know that it's boolean flag. So it's more about code readability than about language syntax :)